### PR TITLE
Adding CreateOrUpdate/PUT option to rest interfaces

### PR DESCRIPTION
**Pull request by allenjzhang**
Also fixed the parameter for POST which should be on the collection. I have left the aggregate interface untouched pending discussion on which one to incorporate.

### CODE REVIEW

**Review by daviwil — APPROVED** [review]
Looks good!  For the tests to past, you need to also check in the updated `openapi.json` for the `rest/petstore` sample.

File: packages/samples/test/output/rest/petstore/openapi.json [hunk 1]
```diff
@@ -124,15 +124,13 @@
             }
           }
         }
-      },
+      }
+    },
+    "": {
```
Comment (nguerrera) [inline packages/samples/test/output/rest/petstore/openapi.json:129]: Is this right?

Comment (daviwil) [reply packages/samples/test/output/rest/petstore/openapi.json:129]: I fixed it in my PR

**Pull request merged**